File: src/main/java/by/dev/madhead/jarvis/util/ChangesFiller.java
package by.dev.madhead.jarvis.util;

import by.dev.madhead.jarvis.creator.ChangeCreator;
import by.dev.madhead.jarvis.model.Change;
import hudson.AbortException;
import hudson.FilePath;
import hudson.scm.ChangeLogSet;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class ChangesFiller {

    private ChangesFiller() {
    }

    public static List<Change> fillChangesList(String gitUrl, FilePath workspace, List<ChangeLogSet<? extends ChangeLogSet.Entry>> changeLogSets) throws AbortException {
        List<Change> changes = new ArrayList<>();
        try {
            ChangeCreator changeCreator = new ChangeCreator(gitUrl, workspace);
            changeLogSets.forEach(changeLogSet ->
                    changeLogSet.forEach(changeLogEntry -> changes.add(changeCreator.create(changeLogEntry))));
        } catch (IOException e) {
            throw new AbortException(e.getMessage());
        }
        return changes;
    }

}

File: src/main/kotlin/by/dev/madhead/jarvis/util/ChangesFiller.kt
package by.dev.madhead.jarvis.util

import by.dev.madhead.jarvis.creator.ChangeCreator
import by.dev.madhead.jarvis.model.Change
import by.dev.madhead.jarvis.model.ChangeSet
import hudson.AbortException
import hudson.FilePath
import hudson.scm.ChangeLogSet
import java.io.IOException

class ChangesFiller {

    fun fillChangesList(gitUrl: String, workspace: FilePath,
                        changeLogSets: List<ChangeLogSet<out ChangeLogSet.Entry>>): ChangeSet {
        val changes = mutableListOf<Change>()
        try {
            val changeCreator = ChangeCreator(gitUrl, workspace)
            changeLogSets.forEach { changeLogSet ->
                changeLogSet.forEach { changeLogEntry ->
                    changes.add(changeCreator.create(changeLogEntry))
                }
            }
        } catch (e: IOException) {
            throw AbortException(e.message)
        }
        return changes
    }

}
